pydicom/valuerep.py

```diff
@@ -344,16 +345,6 @@ def __init__(self, val: Union[str, _TM, datetime.time]) -> None:
             if val.microsecond > 0:
                 self.original_string += f".{val.microsecond:06}"
 
-    if platform.python_implementation() == "PyPy":
-        # Workaround for CPython/PyPy bug in time.__reduce_ex__()
-        #   caused by returning (time, ...) rather than (self.__class__, ...)
-        def __reduce_ex__(self, protocol: int) -> Union[str, Tuple[Any, ...]]:
-            return (
-                self.__class__,
-                super()._getstate(protocol),
-                self.__getstate__()
-            )
-
```

Comment:
@scaramallion - I removed this because the tests pass without it, so I guess this was for old PyPy version. Please check if this is correct.

Reply:
Yeah it got fixed in both CPython and PyPy, if it comes up as an issue we can just tell people to update to a PyPy.